pkg/ccl/backupccl/backup_job.go

```diff
@@ -932,6 +936,28 @@ func (b *backupResumer) Resume(ctx context.Context, execCtx interface{}) error {
 	)
 }
 
+// ensureClusterIDMatches verifies that this job record matches
+// the cluster ID of this cluster.
+// This check ensures that if the backup job has been restored from a
+// backup of a tenant, or from streaming replication, then we will fail
+// this backup since resuming the backup may overwrite backup metadata created by a different cluster.
+func (b *backupResumer) ensureClusterIDMatches(ctx context.Context, clusterID uuid.UUID) error {
+	if createdBy := b.job.Payload().CreationClusterID; createdBy == uuid.Nil {
+		// This cluster was upgraded from a version that did not set a clusterID
```

Comment:
We've been populating this for more than a release now, so I'd just remove this untested branch and just make the mismatch condition be id != nil && id != current.

Reply:
Done.